src/components/vacation-requests-table/vacation-requests-table-toolbar/toolbar-form/toolbar-form-fields.tsx

```diff
@@ -65,26 +75,90 @@ const ToolbarFormFields = ({
     });
   };
 
+  /**
+   * Handle days change
+   *
+   * @param value days value
+   */
+  const handleDaysChange = (value: string) => {
+    const daysValue = Number.parseInt(value) || 0;
+    setVacationRequestData({
+      ...vacationRequestData,
+      days: daysValue
+    });
+  };
+
+  /**
+   * Handle restore default days
+   */
+  const handleRestoreDefaultDays = () => {
+    const defaultDays = calculateTotalVacationDays(dateRange.start, dateRange.end, [
+      true,
+      true,
+      true,
+      true,
+      true,
+      false,
+      false
+    ]);
+
+    setVacationRequestData({
+      ...vacationRequestData,
+      days: defaultDays
+    });
+  };
+
   return (
     <FormControl sx={{ width: "100%" }}>
-      <FormLabel>{strings.vacationRequest.message}</FormLabel>
-      <TextField
-        required
-        error={!vacationRequestData.message?.length}
-        value={vacationRequestData.message}
-        onChange={(event: ChangeEvent<HTMLInputElement>) => {
-          handleVacationRequestDataChange(event.target.value);
-        }}
-        sx={{ marginBottom: "5px" }}
-      />
-      <FormLabel sx={{ marginBottom: "5px" }}>{strings.vacationRequest.days}</FormLabel>
-      <DateRangePicker
-        dateTimeTomorrow={dateTimeTomorrow}
-        dateRange={dateRange}
-        setDateRange={setDateRange}
-      />
+      {!adminMode && (
+        <>
+          <FormLabel>{strings.vacationRequest.message}</FormLabel>
+          <TextField
+            required
+            error={!vacationRequestData.message?.length}
+            value={vacationRequestData.message}
+            onChange={(event: ChangeEvent<HTMLInputElement>) => {
+              handleVacationRequestDataChange(event.target.value);
+            }}
+            sx={{ marginBottom: "5px" }}
+          />
+        </>
+      )}
+
+      {adminMode ? (
+        <>
+          <FormLabel>{strings.vacationRequest.days}</FormLabel>
+          <Box sx={{ display: "flex", alignItems: "flex-start", gap: 1, mb: 1 }}>
+            <TextField
+              type="number"
+              value={vacationRequestData.days}
+              onChange={(event: ChangeEvent<HTMLInputElement>) => {
+                handleDaysChange(event.target.value);
+              }}
+              inputProps={{ min: 1 }}
+              sx={{ flexGrow: 1 }}
+            />
```

Comment:
When testing the UI I notice that when user edits the days and updates the input briefly flashes to show 1 before closing. The update seems to work fine otherwise but would be good if we can fix this brief flash.